pkg/helper/autonode/helpers.go

```diff
@@ -2,28 +2,89 @@ package autonode
 
 import (
 	"fmt"
-	"os"
 	"strings"
 
 	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
 	"github.com/spf13/cobra"
 
 	"github.com/openshift/rosa/pkg/aws"
 	"github.com/openshift/rosa/pkg/interactive"
-	"github.com/openshift/rosa/pkg/interactive/confirm"
 	"github.com/openshift/rosa/pkg/ocm"
 	"github.com/openshift/rosa/pkg/rosa"
 )
 
+// Config holds AutoNode configuration parameters and results
+type Config struct {
+	// Input parameters
+	AutoNodeFlag string // Value from --autonode flag
+	RoleARNFlag  string // Value from --autonode-iam-role-arn flag
+
+	// Output values
+	Mode    string // AutoNode mode to set (e.g., "enabled")
+	RoleARN string // IAM role ARN to use
+}
+
+// ValidateAutoNodeValue validates the autonode flag value
+func ValidateAutoNodeValue(value string) error {
+	if value != "enabled" {
+		return fmt.Errorf("Invalid value for --autonode. Currently only 'enabled' is supported")
+	}
+	return nil
+}
+
+// ValidateRoleARN validates the IAM role ARN format
+func ValidateRoleARN(roleArn string) error {
+	if roleArn == "" {
+		return fmt.Errorf("IAM role ARN cannot be empty")
+	}
+	if !aws.RoleArnRE.MatchString(roleArn) {
+		return fmt.Errorf("Invalid IAM role ARN format: '%s'. Expected format: arn:aws:iam::<account-id>:role/<role-name>", roleArn)
+	}
+	return nil
+}
+
+// ValidateAutoNodeConfiguration validates the overall AutoNode configuration state
+func ValidateAutoNodeConfiguration(autoNodeChanged, roleArnChanged, currentEnabled bool, flagValue, roleArnValue string) error {
+	// Validate enabling AutoNode when already enabled
+	if autoNodeChanged && currentEnabled {
+		return fmt.Errorf("AutoNode is already enabled for this cluster")
+	}
+
+	// Validate IAM role ARN is provided when enabling AutoNode
+	if autoNodeChanged && flagValue == "enabled" && (!roleArnChanged || roleArnValue == "") {
+		return fmt.Errorf("IAM role ARN is required when enabling AutoNode")
+	}
+
+	// Validate can't update IAM role when AutoNode is not enabled
+	if roleArnChanged && !autoNodeChanged && !currentEnabled {
+		return fmt.Errorf("Cannot update IAM role ARN when AutoNode is not enabled. " +
+			"Enable AutoNode first with --autonode=enabled")
+	}
+
+	return nil
+}
+
+// DetermineAutoNodeMode determines the AutoNode mode based on the current state and changes
+func DetermineAutoNodeMode(autoNodeChanged, currentEnabled bool, flagValue string) string {
+	if autoNodeChanged {
+		return flagValue
+	}
+	// If AutoNode is already enabled and only updating the role, maintain current state
+	if currentEnabled {
+		return "enabled"
+	}
```

Comment:
I'm not too sure I see a use for this, we only allow `auto-node` to be `enabled` at the moment. Eventually, this won't be the case though

Either way, all this code seems to do is set the value to `enabled` if the value is already set to `enabled`

So, we're essentially saying `is autonode equal to "enabled?" > set autonode to "enabled"`

Reply:
We also have several checks for `autonode == "enabled"` in each command. Realistically, we only need to validate this one time (after user input is taken)

Reply:
Updated on this commit [9f4c0b0](https://github.com/openshift/rosa/pull/3020/commits/9f4c0b096510fc069cf2531425b3e395b5c4d507)